src/journal/cat.c

```diff
@@ -123,14 +142,23 @@ static int run(int argc, char *argv[]) {
         if (r <= 0)
                 return r;
 
-        fd = sd_journal_stream_fd(arg_identifier, arg_priority, arg_level_prefix);
-        if (fd < 0)
-                return log_error_errno(fd, "Failed to create stream fd: %m");
+        outfd = sd_journal_stream_fd(arg_identifier, arg_priority, arg_level_prefix);
+        if (outfd < 0)
+                return log_error_errno(outfd, "Failed to create stream fd: %m");
+
+        if (arg_stderr_priority == arg_priority) {
+                errfd = outfd;
+        } else {
+                errfd = sd_journal_stream_fd(arg_identifier, arg_stderr_priority, arg_level_prefix);
+                if (errfd < 0)
+                        return log_error_errno(errfd, "Failed to create stream fd: %m");
+        }
 
         saved_stderr = fcntl(STDERR_FILENO, F_DUPFD_CLOEXEC, 3);
 
-        r = rearrange_stdio(STDIN_FILENO, fd, fd); /* Invalidates fd on succcess + error! */
-        TAKE_FD(fd);
+        r = rearrange_stdio(STDIN_FILENO, outfd, errfd); /* Invalidates fd on succcess + error! */
```

Comment:
As described above, I've changed it to `rearrange_stdio(STDIN_FILENO, TAKE_FD(outfd), TAKE_FD(errfd))`, since I think this more closely matches the intended semantics of `TAKE_FD` - it is used to be more explicit about passing ownership of the fd.  Please let me know if you think this is a good way to write it.